### DISCUSSION TITLE
Notification in Admin App that API uses a (part of) collection

### DISCUSSION THREAD

**Discussion by jjvanrossum** [directus/directus#2805] [Archive]
Wouldn't it be so nice to have an end user of the Admin App be aware of the use of collection/fields in the API?

I would think of a notification (in red ofcourse) that when an end user wants to alter the scheme of a collection - the app warns the user about usage elsewhere. Or maybe have it locked even - and having to unlock it on purpose to be able to alter a field/collection.

This way, different teams can work together on the same dataset without having the risk of breaking things when someone just changes something..

Probably something has to be registered to lock a collection from changing - or the Admin App could look at the activity (not sure if there's API-read-activity available..)

**Comment by benhaynes** [comment]
Interesting. I've had the idea of a "schema lock" flag for a while (for prod vs dev). Are you proposing this locks the schema (data model of collections and fields) and/or content itself (items and values)?

**Comment by jjvanrossum** [comment]
Yes, where changes might corrupt other systems extending from directus data, someone might lock the schema or even the data manually.

Another possibility is to require dedicated API users/roles. Whatever collection/field their rights allow to access, will be marked as high risk to alter.. not necessarily locking it, but end user should make extra effort, ensuring they know what they are altering.
